### ISSUE TITLE
Oasis Direct transaction not completing

### ISSUE THREAD

**Issue by john-light** [OasisDEX/oasis#430]
Hello, I recently created an Oasis Direct transaction to buy ETH with DAI. You can see the transaction here:

https://etherscan.io/tx/0xf598a0557e3e472ee9f0f513086deaaa070e649158b0a019eccf8580f71458b1

So you can see that I sent the DAI, and it looks like the ETH got stuck in the middle as WETH in the address `0x5ae9c9e696549e5508e18848413ef26710edadad` and I never got the ETH in my address `0xd977dA63d086d222EDE0aa68ee84328310485FFE`.

Do you know what went wrong and when I will get my ETH?

Thanks

**Comment by pimato** [comment]
you got your eth, just look on the tab "internal transactions" on etherscan: 

<img width="1161" alt="bildschirmfoto 2018-08-15 um 17 59 04" src="https://user-images.githubusercontent.com/5337809/44158454-faf94e00-a0b4-11e8-90c4-efd9cb49e43d.png">

**Comment by john-light** [comment]
@pimato do you know how I actually access that ETH? It isn't showing up in Metamask or MyCrypto. Including that ETH my balance should be somewhere around 0.95 ETH but right now it's showing as 0.767602832715987654 ETH and there's no WETH or PETH in the account either.

**Comment by gbalabasquer** [comment]
@john-light 
So if you check the address `0x5ae9c9e696549e5508e18848413ef26710edadad` (which is your proxy account) doesn't have any WETH stuck: https://etherscan.io/token/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2?a=0x5ae9c9e696549e5508e18848413ef26710edadad

And here you have the transactions that prove the paths:
https://etherscan.io/address/0x5ae9c9e696549e5508e18848413ef26710edadad#tokentxns
It shows how your proxy got DAI from you, sent it to the owner of the order that got matched and received WETH from the oasis contract.

Then you have your account:
https://etherscan.io/address/0xd977da63d086d222ede0aa68ee84328310485ffe#internaltx
If you see the internal transactions, you will see your proxy sent you native ETH for the same amount that received of WETH.

So I've added all your ETH inputs and outputs in a spreadsheet (including this internal transaction) and the final value is 0.7958509423. If you take into account the gas fees paid, 0.76... is just fine, not sure why you think it should be around 0.95.

**Comment by john-light** [comment]
@gbalabasquer thanks so much for looking further into this! I did the same exercise (accounting my balance) and noticed that on the same day, at almost the same time as the Oasis Dex tx there was a tx in the same amount (0.2) that went to another wallet. Hence why I had expected my wallet to have 0.2 ETH more in it. I do recall now making that transaction as a test between MetaMask and MyCrypto desktop wallet before testing Oasis Dex but clearly forgot and didn't notice it when reviewing the tx history in Etherscan.

So, in short, I apologize for wasting reviewers' time here and thank you all for trying to help me out 😅
